src/notificationcenter.cs

```diff
@@ -103,14 +103,21 @@ interface NSExtensionContext_NCWidgetAdditions {
 	[BaseType (typeof (UIVibrancyEffect))]
 	interface UIVibrancyEffect_NCWidgetAdditions {
 		[iOS (10,0)]
+		[Deprecated (PlatformName.iOS, 13,0, message: "Use 'UIVibrancyEffect.GetWidgetEffect' instead.")]
 		[Static]
 		[Export ("widgetPrimaryVibrancyEffect")]
 		UIVibrancyEffect GetWidgetPrimaryVibrancyEffect ();
 
 		[iOS (10,0)]
+		[Deprecated (PlatformName.iOS, 13,0, message: "Use 'UIVibrancyEffect.GetWidgetEffect' instead.")]
 		[Static]
 		[Export ("widgetSecondaryVibrancyEffect")]
 		UIVibrancyEffect GetWidgetSecondaryVibrancyEffect ();
+
+		[iOS (13,0)]
+		[Static]
+		[Export ("widgetEffectForVibrancyStyle:")]
+		UIVibrancyEffect GetWidgetEffect (UIVibrancyEffectStyle vibrancyStyle);
```

Comment:
The changes are correct but incomplete. This generated type is `internal` (see `[Internal]` at the type level) so the deprecation and the new API won't be visible to developers.

That's because they are `static` methods inside a category - which in C# are bound as (static) extension methods... conflicting with adding static (non extensions) methods.

To expose the new API (and deprecation) correctly you need to update the `src/UIKit/UIVibrancyEffect.cs` file too

Reply:
So @joconte would just have to add the GetWidgetEffect method into the UIVibrancyEffect.cs file? Also why is the file corresponding to the NotificationCenter in the UIKit folder instead of the NotificationCenter folder? Just wondering so I will know where to check in this kind of situation also.

Reply:
@tj-devel709 It's an ObjC category which, in C#, is _most similar_ to extension methods - but we can't use them (at least nicely) as such as they are (ObjC) `static` methods. 
Still the category/extension is done on an UIKit type so it's why the final (manual) bits of code will reside inside UIKit (because it's how it will be most discoverable, on `UIVibrancyEffect`).

Reply:
@tj-devel709 Thanks for asking for clarification BTW. The answer from @spouliot helped me too.